code/modules/mob/living/carbon/species.dm

```diff
@@ -938,11 +939,36 @@ var/list/has_died_as_golem = list()
 					dustmind.current = G
 					mind = null
 					G.key = key
-					to_chat(G, "You are an adamantine golem. You move slowly, but are highly resistant to heat and cold as well as impervious to burn damage. You are unable to wear most clothing, but can still use most tools. Serve [user], and assist them in completing their goals at any cost.")
+					to_chat(G, "You are an [G.name]. You move slowly, but are highly resistant to heat and cold as well as impervious to burn damage. You are unable to wear most clothing, but can still use most tools. Serve [user], and assist them in completing their goals at any cost.")
```

Comment:
`\an [G]` perhaps? Else it would read "You are an Clockwork Golem" or something?